app/services/katello/registration_manager.rb

```diff
@@ -264,12 +264,23 @@ def set_host_collections(host, activation_keys)
         host.host_collection_ids = host_collection_ids
       end
 
-      def lookup_content_view_environment(activation_keys)
+      def lookup_content_view_environments(activation_keys)
+        # If the setting is on, we combine all CVEs from all AKs
+        if Setting['allow_multiple_content_views']
+          cves = activation_keys.map do |act_key|
+            act_key.content_view_environments
+          end
+          cves = cves.flatten.uniq
+          fail _('At least one activation key must have a lifecycle environment and content view assigned to it') if cves.blank?
+          return cves
+        end
+
+        # If the setting is off, we stick with the previous behavior (the last AK with a valid cv/lce wins).
         activation_key = activation_keys.reverse.detect do |act_key|
-          act_key.environment && act_key.content_view
+          act_key.content_view_environments.any?
         end
         if activation_key
-          ::Katello::ContentViewEnvironment.where(:content_view_id => activation_key.content_view, :environment_id => activation_key.environment).first
+          ::Katello::ContentViewEnvironment.where(:content_view_id => activation_key.content_view, :environment_id => activation_key.environment)
```

Comment:
If I turn off the Setting, that will cause this codepath to be reached. If, however, I am still trying to register with a multi-CV AK, this results in a successful registration with the host getting only the first CVE on that AK (because `.content_view` and `.environment` returns the first.) Instead, registration should fail with an error.

This is part of the reason I avoided keeping those methods on content facet, and instead forced myself to use `single_content_view` and `single_lifecycle_environment` - you can't do that without intending to do it. After fixing the issue here, I would recommend removing the `content_view` and `environment` methods from activation key.

Reply:
> If I turn off the Setting, that will cause this codepath to be reached. If, however, I am still trying to register with a multi-CV AK, this results in a successful registration with the host getting only the first CVE on that AK (because `.content_view` and `.environment` returns the first.) Instead, registration should fail with an error.

Updated in the latest commits.